src/main/java/uk/gov/justice/hmpps/datacompliance/services/migration/OffenderIterator.java

```diff
@@ -1,27 +1,38 @@
 package uk.gov.justice.hmpps.datacompliance.services.migration;
 
-import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 import uk.gov.justice.hmpps.datacompliance.config.DataComplianceProperties;
 import uk.gov.justice.hmpps.datacompliance.dto.OffenderNumber;
 import uk.gov.justice.hmpps.datacompliance.services.client.elite2api.Elite2ApiClient;
 import uk.gov.justice.hmpps.datacompliance.services.client.elite2api.Elite2ApiClient.OffenderNumbersResponse;
 
+import java.util.concurrent.ExecutorService;
 import java.util.function.Consumer;
 import java.util.stream.LongStream;
 
 import static java.lang.Long.MAX_VALUE;
 import static java.lang.Math.ceil;
 import static java.lang.Math.min;
+import static java.util.concurrent.Executors.callable;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.stream.Collectors.toList;
+import static uk.gov.justice.hmpps.datacompliance.utils.Exceptions.propagateAnyError;
 
 @Slf4j
 @Service
-@AllArgsConstructor
 class OffenderIterator {
 
     private final Elite2ApiClient elite2ApiClient;
     private final DataComplianceProperties properties;
+    private final ExecutorService executorService;
+
+    public OffenderIterator(final Elite2ApiClient elite2ApiClient,
+                            final DataComplianceProperties properties) {
+        this.executorService = newFixedThreadPool(properties.getElite2ApiOffenderIdsIterationThreads());
```

Comment:
Isn't there the standard pool (sorry can't remember name  of it), the one that is based on the number of processors which would be used by default? Not sure the advantage of say 10 threads would be if you only have say 2 processors?